src/common/datasource/src/buffered_writer.rs

```diff
@@ -79,8 +73,15 @@ impl<
         Fut: Future<Output = Result<T>>,
     > LazyBufferedWriter<T, U, F>
 {
-    /// Closes the writer without flushing the buffer data.
+    /// Closes the writer and flushes the buffer data.
```

Comment:
This method has only two invocations. We can do this refactor and keep `try_flush` private.